source/common/upstream/outlier_detection_impl.cc

```diff
@@ -818,6 +911,11 @@ void EventLoggerImpl::logEject(const HostDescriptionConstSharedPtr& host, Detect
             : DetectorHostMonitor::SuccessRateMonitorType::LocalOrigin;
     event.mutable_eject_failure_percentage_event()->set_host_success_rate(
         host->outlierDetector().successRate(monitor_type));
+  } else if (type == envoy::data::cluster::v3::EXTENSION) {
+    event.mutable_eject_extension_event()->set_monitor_name(
+        host->outlierDetector().getFailedExtensionMonitorName());
+    event.mutable_eject_extension_event()->set_extra_info(
+        host->outlierDetector().getFailedExtensionMonitorExtraInfo());
```

Comment:
How about the case where multiple extensions are failed?

Reply:
Same as above. Will rethink how to pass more than one monitor.

Reply:
It is safe now. If several monitors fail at the same time, multiple events will be posted from worker threads to the main thread. As explained above, only one such event will be processed, the rest will be ignored. I also changed the code, so the name of the monitor is passed via event to the main thread and the monitor does not need to be queried for name. Extra info has been removed, as it is not used by consecutive errors monitor.